backend/hexa/pipelines/schema/mutations.py

```diff
@@ -652,6 +653,40 @@ def resolve_add_pipeline_output(_, info, **kwargs):
     return {"success": True, "errors": []}
 
 
+@pipelines_mutations.field("updatePipelineHeartbeat")
+def resolve_update_pipeline_heartbeat(_, info, **kwargs):
+    request: HttpRequest = info.context["request"]
+    if not request.user.is_authenticated or not isinstance(
+        request.user, PipelineRunUser
+    ):
+        return {
+            "success": False,
+            "errors": ["PIPELINE_NOT_FOUND"],
+        }
+
+    try:
+        pipeline_run = PipelineRun.objects.get(pk=request.user.pipeline_run.id)
+    except PipelineRun.DoesNotExist:
+        return {
+            "success": False,
+            "errors": ["PIPELINE_NOT_FOUND"],
```

Comment:
I feel like we should make sure an error like this ends up in Sentry since it would most likely be a bug if this actually occurs. What do you think?